pkg/server/container_create_windows.go

```diff
@@ -517,18 +513,57 @@ func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxP
 		}
 	}
 
+	// hyperv pci devices are exposed as windows devices on the container spec
+	// given to hcsshim.
+	addOCIWindowsDevices(&g, rhcso, config.Devices)
+
 	return g.Config, nil
 }
 
-func addOCIWindowsDevices(g *generator, devs []*runtime.Device) {
+// getWindowsDeviceInfo is a helper function that returns a spec specified device's
+// prefix and device identifier. A prefix is any string before "://" in the spec
+// device's `HostPath`.
+func getWindowsDeviceInfo(hostPath string) (string, string) {
+	substrings := strings.SplitN(hostPath, "://", 2)
+	if len(substrings) <= 1 {
+		return "", ""
+	}
+	return substrings[0], substrings[1]
+}
+
+// validWindowsDeviceForIsolation is a helper function that returns if a given windows
+// device prefix is valid for the isolation type of the container.
+//
+// Hyper-V isolated containers only support the prefixes "gpu" and "vpci".
+//
+// Process isolated containers only support the prefixes "vpciLocationPath"
```

Comment:
Some information on what these types mean (or where I can find that info) would be good here.

Reply:
They're pretty arbitrary and since our assigned devices wiki is not public, I'll just update here :)